### PR TITLE
feat: allow external config incl. feature targeting overrides

### PR DESCRIPTION
**Pull request by roncohen**
- Introduce comprehensive configuration options from environment variables and a local file.
- Introduced feature targeting overrides which are useful in testing and with local development.
- Switch logging to be `info` by default.

### CODE REVIEW

File: packages/node-sdk/README.md [hunk 1]
```diff
@@ -100,6 +100,38 @@ to `getFeatures` (or through `bindClient(..).getFeatures()`). That means the
...
+| `secretKey`        | string                  | The secret key used for authentication with Bucket's servers.                                                          | BUCKET_SECRET_KEY                               |
+| `logLevel`         | string                  | The log level for the SDK (e.g., `"info"`, `"warn"`, `"error"`). Default: `info`                                       | BUCKET_LOG_LEVEL                                |
+| `offline`          | boolean                 | Operate in offline mode. Default: `false`, except in tests it will default to `true` based off of the `TEST` env. var. | BUCKET_OFFLINE                                  |
+| `host`             | string                  | The host URL for the Bucket servers.                                                                                   | BUCKET_HOST                                     |
+| `featureOverrides` | Record<string, boolean> | An object specifying feature overrides for testing or local development                                                | BUCKET_FEATURE_ENABLED, BUCKET_FEATURE_DISABLED |
```
Comment (matus-vacula) [inline packages/node-sdk/README.md:114]: you're using plural in the code. BUCKET_FEATURES_ENABLED and BUCKET_FEATURES_DISABLED

File: packages/node-sdk/README.md [hunk 2]
```diff
@@ -100,6 +100,38 @@ to `getFeatures` (or through `bindClient(..).getFeatures()`). That means the
...
+
+Note: BUCKET_FEATURE_ENABLED, BUCKET_FEATURE_DISABLED are comma separated lists of features which will be enabled or disabled respectively.
```
Comment (matus-vacula) [inline packages/node-sdk/README.md:116]: same here

File: packages/node-sdk/README.md (hunk 1 above)
Comment (matus-vacula) [reply packages/node-sdk/README.md:114]: Let's also add BUCKET_CONFIG_FILE

Comment (roncohen) [reply packages/node-sdk/README.md]: Good catch!

File: packages/node-sdk/README.md (hunk 2 above)
Comment (roncohen) [reply packages/node-sdk/README.md]: thanks!

**Review by matus-vacula — APPROVED** [review]

**Pull request merged**